server/src/models/people/populate.js

```diff
@@ -1,46 +1,87 @@
 import _ from "lodash";
 
-import { people_years } from "../constants.js";
 import { get_standard_csv_file_rows } from "../load_utils.js";
 
-import {
-  camel_case_headcount_model_names,
-  snake_case_headcount_model_names,
-} from "./headcount_model_utils.js";
-
-const headcount_csv_names_by_headcount_model_name = _.chain(
-  camel_case_headcount_model_names
-)
-  .zip(snake_case_headcount_model_names)
-  .fromPairs()
-  .value();
-
-const format_csv_records_then_register_to_headcount_models = (
-  headcount_models_and_records
-) => {
-  _.each(headcount_models_and_records, (model_and_record) => {
-    _.each(model_and_record.records, (record) =>
-      _.each(
-        people_years,
-        (col) => (record[col] = record[col] && parseFloat(record[col]))
-      )
-    );
-    _.each(model_and_record.records, (record) =>
-      model_and_record.model.register(record)
-    );
-  });
-};
-
-export default function ({ models }) {
-  const headcount_models_and_records = _.map(
-    headcount_csv_names_by_headcount_model_name,
-    (headcount_csv_name, headcount_model_name) => ({
-      model: models[headcount_model_name],
-      records: get_standard_csv_file_rows(`${headcount_csv_name}.csv`),
-    })
-  );
-
-  format_csv_records_then_register_to_headcount_models(
-    headcount_models_and_records
-  );
+export default async function ({ models }) {
+  const {
+    EmployeeAgeGroup,
+    EmployeeAvgAge,
+    EmployeeExLvl,
+    EmployeeFirstOfficialLang,
+    EmployeeGender,
+    EmployeeRegion,
+    EmployeeType,
+  } = models;
+
+  let employee_age_rows,
+    employee_avg_age_rows,
+    employee_ex_lvl_rows,
+    employee_fol_rows,
+    employee_gender_rows,
+    employee_region_rows,
+    employee_type_rows;
+  const process_employee_csv = (csv_name) =>
+    _.chain(get_standard_csv_file_rows(csv_name))
+      .reject(["dept_code", "ZGOC"])
+      .map(({ dept_code, dimension, ...data_columns }) => ({
+        dept_code,
+        dimension,
+        ..._.mapValues(data_columns, _.toNumber),
+      }))
+      .groupBy("dept_code")
+      .map((dept_rows, dept_code) => ({
+        dept_code,
+        data: _.chain(dept_rows)
+          .map(({ dept_code, dimension, avg_share, ...data_columns }) => ({
```

Comment:
Small nitpick, on line 26 `avg_share` is included in the thing called `data_columns`, but on line 35 it is excluded. Best to try not reusing names for things that are different in subtle ways like this. Maybe `numeric_columns` the first time and `values_by_year` the second time?